rootfs/apply.go

```diff
@@ -0,0 +1,136 @@
+package rootfs
+
+import (
+	"io"
+	"io/ioutil"
+
+	"github.com/docker/containerd"
+	"github.com/docker/containerd/log"
+	"github.com/docker/distribution"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type Snapshotter interface {
```

Comment:
@AkihiroSuda We use narrow interfaces for input. By defining here, we avoid coupling these packages to each other. We'll just call this a `Snapshotter`.

The `SnapshotManager` is likely to be a concrete type that uses `SnapshotDriver`s to implement `Snapshotter`.